net/WebSocketHandler.hpp

```diff
@@ -241,21 +246,52 @@ class WebSocketHandler : public ProtocolHandlerInterface
     void shutdown(const StatusCodes statusCode = StatusCodes::NORMAL_CLOSE,
                   const std::string& statusMessage = std::string(),
                   bool hardShutdown = false)
+    {
+        shutdownImpl(statusCode, statusMessage, hardShutdown, false);
+    }
+
+    /// Don't wait for the remote Websocket to handshake with us; go down fast.
+    void shutdownAfterWriting()
+    {
+        shutdownImpl(WebSocketHandler::StatusCodes::NORMAL_CLOSE, std::string(),
+                     true /* hard async shutdown & close */, false);
+    }
+
+    /// Returns true if the underlying socket is connected.
+    bool isConnected() const {
+        std::shared_ptr<StreamSocket> socket = _socket.lock();
+        return nullptr != socket && !socket->isClosed();
```

Comment:
Please, please, don't do reverse comparisons. It's very unnatural and an eye-sore. 

`return socket && !socket->isClosed();` OR `return socket != nullptr && !socket->isClosed();`

Reply:
While not always doing it, it is a good way to avoid accidental assignment and showing the r-value on the left side. 
So being trained this way, it is natural. However, I also agree that compiler and tools to warn against such accidents. 
Hmm .. but will keep your preference in mind. No guarantees.

Reply:
We write once, read many, many times.

Double checking what we write is preferable to slowing down all readers just so we won't make accidental mistakes. I understand why some make such suggestions to beginners (they want to help them avoid bugs). As a teacher, I rather teach to improve quality and skills, rather than tricks to avoid sloppy coding. That's not helping the beginner to get any better (having fewer bugs doesn't make one better).

But the burden on the reader is huge. It forces the reader to remember what is being compared against before knowing what the subject of the comparison is. So when scanning code (something we do all the time, far more than typing code), we can't quickly trace the variables we are interested in, and we have to stop and reason about `0 != value` and `nullptr != pointer` etc. to decide 1) do I care about this statement, 2) which variable is involved? 3) what is it doing?

There is a reason each language has idiomatic style; it makes it easier to read code quickly and more accurately. That's why we use `!` for negation, instead of `== false`, for example.